hardhat/env-artifacts.js

```diff
@@ -1,5 +1,9 @@
 const { HardhatError } = require('hardhat/internal/core/errors');
 
+process.on('unhandledRejection', reason => {
+  throw new Error(reason);
```

Comment:
I couldn't find anything useful in the second argument of the listener (the promise). It doesn't give the stacktrace anyway and this one works fine.

The only drawback is that it halts the tests whenever a failure is found. We can tweak it to behave differently but I'm personally fine with it.

Reply:
I wouldn't have put it in this file. IMO this should be its own file in `hardhat/`.

Maybe `hardhat/test-sanity.js`

Reply:
I'm fine with `test-sanity.js`